frodo.js

```diff
@@ -6,127 +6,129 @@
  *
```

Comment:
SCRIPT-I'm a bit concerned about the potential performance impact of this loop. Have you thought about optimizing it?